api/src/main/java/org/apache/iceberg/SnapshotUpdate.java

```diff
@@ -71,4 +71,8 @@ default ThisT toBranch(String branch) {
             "Cannot commit to branch %s: %s does not support branch commits",
             branch, this.getClass().getName()));
   }
+
+  default String targetBranch() {
+    return SnapshotRef.MAIN_BRANCH;
+  }
```

Comment:
We should probably separate this API PR if we determine this is the right way to solve this issue

Reply:
I think it's fine to keep it in the same PR if we need to add this API to fix the logic, it makes the purpose more clear.